feat: add no-cache support

innovaker opened the pull request:
`no-cache` rebuilds everything from scratch.  This is critical for integrating security patches and bug fixes as soon as possible.

Requires: #64

petejohanson commented on .github/workflows/containers.yml:
```diff
@@ -268,9 +289,10 @@ jobs:
-            type=registry,ref=${{ steps.paths.outputs.build-candidate }}
-            type=registry,ref=${{ steps.paths.outputs.build-branch }}
-            ${{ (steps.paths.outputs.build-base != '') && format('type=registry,ref={0}', steps.paths.outputs.build-base) || '' }}
+            ${{ (env.no-cache == 'false') && format('type=registry,ref={0}', steps.paths.outputs.build-candidate) }}
```
No way to add this check once, around the whole group?

innovaker replied:
You're right, it's a little ungainly.  I've pushed an alternative.  Thoughts?

innovaker replied:
I've looked into this and my impression is that the GitHub Actions expressions syntax is quite limited.  It doesn't *appear* to support inline concatination or array syntax (making `join` seemingly impractical).  Have you any suggestions?

I suspect it would require additional shell steps.

innovaker replied:
I've been tentatively planning a refactor for post-2.6.0 which would probably ease this issue ...

- rename existng `paths` step as `tags`
- define new `paths` step which outputs everything in the form `type=registry/local,ref=${{ tag }}`
- list `paths` within `cache-from` and `cache-to` (similar to current approach)

... so perhaps we can leave this until then?

petejohanson replied:
Yeah, that's fine. leaving for now isn't a problem.

petejohanson commented on .github/workflows/containers.yml:
```diff
@@ -287,9 +309,10 @@ jobs:
-            type=registry,ref=${{ steps.paths.outputs.dev-candidate }}
-            type=registry,ref=${{ steps.paths.outputs.dev-branch }}
-            ${{ (steps.paths.outputs.dev-base != '') && format('type=registry,ref={0}', steps.paths.outputs.dev-base) || '' }}
+            ${{ (env.no-cache == 'false') && format('type=registry,ref={0}', steps.paths.outputs.dev-candidate) }}
```
Ditto?

petejohanson reviewed (approved):
Looks fine, but a couple questions on implementation.

innovaker committed:
feat: add no-cache support

`no-cache` rebuilds everything from scratch.  This is critical for integrating security patches and bug fixes as soon as possible.

PR: #65

innovaker commented:
Removed `no-cache-guard` because it was hangover from an earlier design and no longer necessary.
